syft/core/workers.py

```diff
@@ -138,7 +188,35 @@ def receive_msg(self, message_wrapper):
           of object types. However, the object is typically only used during testing or
           local development with :class:`VirtualWorker` workers.
         """
-        raise NotImplementedError
+
+        if(is_binary):
```

Comment:
How does the remote worker know ahead of time if it's binary?  This seems like a good place for try/except

Reply:
Not sure. Copy pasted from socket worker. Everything is guaranteed to be binary. I think we can remove that